modules/system/classes/MediaLibrary.php

```diff
@@ -834,4 +898,51 @@ protected function generateRandomTmpFolderName($location)
 
         return $tmpPath;
     }
+
+    /**
+     * Generates a incremental file name based
+     * on the existing files in the same folder.
+     *
+     * @param string $path Specifies a file path to check.
+     * @return string The generated file name.
+     */
+    public function generateIncrementedFileName($path): string
+    {
+        $pathInfos = pathinfo($path);
+        $dirName = dirname($this->getMediaPath($path));
+
+        $sameFilesInFolder = Arr::map(array_filter(
+            $this->getStorageDisk()->files($dirName),
+            function ($file) use ($pathInfos) {
+                return preg_match('/'. $pathInfos['filename'] .'(_(\d*))?\.'. $pathInfos['extension'] .'$/U', $file);
+            }
+        ), function ($value) use ($dirName) {
+            return str_replace($dirName .'/', '', '/'. $value);
+        });
+
+        return File::unique($pathInfos['basename'], $sameFilesInFolder);
```

Comment:
Is the filtering necessary here or can we just send the list of filenames in the folder unfiltered?

Reply:
You're right, filtering on file name and extension was not required.